pkg/skaffold/build/local/types.go

```diff
@@ -111,6 +112,11 @@ func NewBuilder(cfg Config) (*Builder, error) {
 	}, nil
 }
 
+func (b *Builder) WithArtifactStore(store build.ArtifactStore) *Builder {
```

Comment:
With the exception of some test classes, we seem to use With* as a builder pattern, where the function returns a new object instead of a pointer.  Maybe just `ArtifactStore(store)` or `RecordArtifacts(store)`?

Reply:
I also thought it looked funny. The return type isn't really required.

Reply:
done.